core/services/gateway/connectionmanager.go

```diff
@@ -202,6 +205,12 @@ func (m *connectionManager) FinalizeHandshake(attemptId string, response []byte,
 	if err != nil || attempt.nodeAddress != "0x"+hex.EncodeToString(signer) {
 		return network.ErrChallengeInvalidSignature
 	}
+	if conn != nil {
+		conn.SetPongHandler(func(data string) error {
+			m.lggr.Debugw("received heartbeat pong from node", "nodeAddress", attempt.nodeAddress)
```

Comment:
Isn't this excessive logging? Maybe downgrade it to Trace, or just don't do?

Reply:
Ping/pong will be sent only about every 30 sec.
But your right that a prom counter should suffice. I'll change it when adding prom counters everywhere.